pkg/e2e/infra/infra.go

```diff
@@ -123,40 +121,74 @@ func invalidMachinesetWithEmptyProviderConfig() *mapiv1beta1.MachineSet {
 	}
 }
 
+func buildMachineSetParams(client runtimeclient.Client, replicas int) e2e.MachineSetParams {
+	// Get the current workers MachineSets so we can copy a ProviderSpec
+	// from one to use with our new dedicated MachineSet.
+	workers, err := e2e.GetWorkerMachineSets(client)
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	providerSpec := workers[0].Spec.Template.Spec.ProviderSpec.DeepCopy()
+	clusterName := workers[0].Spec.Template.Labels[e2e.ClusterKey]
+
+	msName := e2e.RandomString()
+
+	return e2e.MachineSetParams{
+		Name:         msName,
+		Replicas:     int32(replicas),
+		ProviderSpec: providerSpec,
+		Labels: map[string]string{
+			"mhc.e2e.openshift.io": msName,
+			e2e.ClusterKey:         clusterName,
+		},
+	}
+}
+
 var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
 	defer g.GinkgoRecover()
 
-	deleteObject := func(client runtimeclient.Client, obj runtime.Object) error {
-		cascadeDelete := metav1.DeletePropagationForeground
-		return client.Delete(context.TODO(), obj, &runtimeclient.DeleteOptions{
-			PropagationPolicy: &cascadeDelete,
-		})
-	}
+	var client runtimeclient.Client
+	var machineSet *mapiv1beta1.MachineSet
 
-	g.It("have machines linked with nodes", func() {
+	g.BeforeEach(func() {
 		var err error
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
 
-		o.Expect(isOneMachinePerNode(client)).To(o.BeTrue())
-		o.Expect(err).NotTo(o.HaveOccurred())
+		client, err = e2e.LoadClient()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		machineSetParams := buildMachineSetParams(client, 3)
+
+		g.By("Creating a new MachineSet")
+		machineSet, err = e2e.CreateMachineSet(client, machineSetParams)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		e2e.WaitForMachineSet(client, machineSet.GetName())
+	})
+
+	g.AfterEach(func() {
+		g.By("Deleting the new MachineSet")
+		err := client.Delete(context.Background(), machineSet)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		e2e.WaitForMachineSetDelete(client, machineSet)
+	})
+
+	g.It("have machines linked with nodes", func() {
+		selector := machineSet.Spec.Selector
```

Comment:
I actually don't think we need this test anymore since this is validated every time we `WaitForMachineSet` beforeEach test